### ISSUE TITLE
False-positive "unresolved symbol" error in dynamic map destructuring

### ISSUE THREAD

**Issue by bfontaine** [borkdude/clj-kondo#1011]
**version**

v2020.09.09

**platform**

native

**problem**

clj-kondo incorrectly reports as an error a map binding that uses a previous binding for its key, e.g.:
```clojure
{a :a  ;; bind a to what’s in the key :a 
 b a} ;; bind b to what’s in the key 'the value of a'
```

**repro**

```clojure
(ns repro)

(defn foo
  [{a :a
    b a}]
  (prn a b))

(foo {:a         :something
      :something 42})
```

```
$ clj repro.clj 
:something 42

$ clj-kondo --lint repro.clj 
repro.clj:5:7: error: unresolved symbol a
linting took 9ms, errors: 1, warnings: 0
```

**config**

none

**expected behavior**

No error.

**Comment by borkdude** [comment]
This is a duplicate of #782 and will be warned against in #916.

**Comment by borkdude** [comment]
Not exactly a duplicate, but certainly related. The reason this is undefined is because maps are un-ordered.

**Comment by bfontaine** [comment]
Thanks; I didn’t saw it.
